ecosystem/sep-0007.md

```diff
@@ -36,6 +36,7 @@ The `tx` operation represents a request to sign a specific XDR `TransactionEnvel
 1. `callback` (optional) - If this value is omitted then the URI handler should sign the given XDR and submit it to the network. If the value is present then it should be interpreted as a URL-encoded callback. The URL-encoded callback will start with a prefix to denote whether this is a `url` callback or some other form of callback. In the case where it is a url callback (denoted by `url:`, i.e. `url%3A` after encoding) the URI handler should send the signed XDR to this url in a `POST` request with `Content-Type` set to `application/x-www-form-urlencoded` with the data fields `xdr` containing the signed XDR (URL-encoded). If there are any query params specified in the URL callback then those should be included in the URL when submitting. For now only `url` callback types are supported.
 1. `pubkey` (optional) - Specify which public key you want the URI handler to sign for. Useful with the `callback` parameter above for example with multisig coordination.
+1. `chain` (optional) -  There can be an optional `chain` query param to include a single SEP-0007 request that spawned or triggered the creation of this SEP-0007 request. This will be a URL-encoded value. The goal of this field is to be informational only and can be used in the case of forwarding SEP-0007 requests. The embedded SEP-0007 request of course can itself contain another embedded SEP-0007 request in its own `chain` field, but caution should be exercised when doing so since this can easily explode the length of the final SEP-0007 request. To prevent attacks against this field, wallets can expect this field to have no more than 7 nested levels.
```

Comment:
Could you add an example of how this would be used, and what purpose it would serve?

>The goal of this field is to be informational only

Informational in what way? Who would view the field, and how would they use it?

If the goal is to link requests, we could introduce the concept of a hash or an identifier. This will be easier to use to reconcile across in other systems like logs.

Reply:
will add a blurb on this.
using hashes would be limiting because some or all of the SEP-7 requests could be internal-only to the systems that spawned the given request.

in the future if we want to add support for hashes we can always do that since this format is forward compatible, for example:
`chain=web+stellar:` vs `chain=hash:`